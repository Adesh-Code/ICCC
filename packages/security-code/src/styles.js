const styles = ({ tokens, primitives }) => ({
  "&.DhigSecurityCode": {
    width: "56px",
    height: "72px",
    borderTop: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`,
    borderBottom: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"][class*="MuiInputBase-inputAdornedEnd"]':
    {
      marginRight: "0"
    },
  "&.DhigSecurityCode[class*='MuiInput-root']:hover:before": {
    height: "2px",
    backgroundColor: tokens.colors["fill-engage"],
    boxShadow: "none",
    borderBottom: `${primitives.borderWidths.medium} solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode:hover": {
    boxShadow: "none"
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.primary.white}`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode input:focus": {
    boxShadow: `0px 2px 0px 0px ${tokens.colors["ink-focus"]} inset, 0px 0px 0px 2px ${tokens.colors["ink-focus"]} inset`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:not(.Mui-disabled)::before': {
    boxShadow: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]:before': {
    width: "56px"
  },
  "&.DhigSecurityCode input": {
    flexGrow: "1"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "56px",
    height: "72px",
    fontSize: `${primitives.fontSizeFallbacks["6"]}`,
    fontWeight: primitives.fontWeights.default
  },
  '&.DhigSecurityCode--dense [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "32px",
    height: "40px",
    fontSize: `${primitives.fontSizeFallbacks["3"]}`,
    fontWeight: primitives.fontWeights.default
  },
  "&.DhigSecurityCode--dense": {
    width: "32px",
    height: "40px"
  },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode--error, &.DhigSecurityCode:first-of-type.DhigSecurityCode--error, &.DhigSecurityCode:last-child.DhigSecurityCode--error, &.DhigSecurityCode--error:before":
    {
      borderColor: primitives.colors.accent.clay
    },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:after': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode:first-of-type": {
    borderLeft: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child": {
    borderRight: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child [class*='MuiDivider-root'], &.DhigSecurityCode:focus-within [class*='MuiDivider-root']":
    {
      display: "none"
    }
});

export default styles;
