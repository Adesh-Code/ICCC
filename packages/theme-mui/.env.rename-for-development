# To load the MUI X Pro license for local development, rename this file to .env and replace with the actual license string
DHIG_MUIXPRO_ENTRY=PASTE_MUI_X_PRO_LICENSE_STRING_HERE