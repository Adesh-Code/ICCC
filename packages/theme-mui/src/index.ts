// Themes
export { default as themeDark } from "./themes/dark";
export { default as themeLight } from "./themes/light";

// Utilities
export { configureClassnameGenerator } from "./classname-generator";
export { setLicenseKey } from "./set-license-key";
