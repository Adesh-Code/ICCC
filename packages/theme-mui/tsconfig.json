{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib/esm",
    "resolveJsonModule": true,
    "declarationDir": "lib/types"
  },
  "include": ["src"]
}
