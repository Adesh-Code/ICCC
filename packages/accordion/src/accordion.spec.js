import React from "react";
import { render } from "@testing-library/react";
import Accordion from "./accordion";
import Typography from "@iccc/typography";

import { themeLight } from "@iccc/theme-mui";
import { createTheme, ThemeProvider } from "@mui/material/styles";

const light = createTheme(themeLight);

const arrange = ({ ...otherProps } = {}) =>
  render(
    <ThemeProvider theme={light}>
      <Accordion {...otherProps}>
        <Typography className="dhig-m-0">Accordion content</Typography>
      </Accordion>
    </ThemeProvider>
  );

describe("Accordion", () => {
  it("mounts an accordion", () => {
    const { container } = arrange();
    expect(container).toMatchSnapshot();
  });

  it("renders the dense variant", () => {
    const { container } = arrange({ dense: true });
    const accordion = container.firstChild;
    expect(accordion).toHaveClass("DhigAccordion--density--dense");
  });

  it("renders the minimal variant", () => {
    const { container } = arrange({ variant: "minimal" });
    const accordion = container.firstChild;
    expect(accordion).toHaveClass("DhigAccordion--variant--minimal");
  });

  it("renders the normal variant using the 'size' prop", () => {
    const { container } = arrange({ size: "normal" });
    const accordion = container.firstChild;
    expect(accordion).not.toHaveClass("DhigAccordion--density--dense");
  });

  it("renders the flat variant using the 'styleVariant' prop", () => {
    const { container } = arrange({ styleVariant: "flat" });
    const accordion = container.firstChild;
    expect(accordion).toHaveClass("DhigAccordion--variant--flat");
  });

  it("disables transitions", () => {
    arrange({ disableTransitions: true });
    // not tested because transitions are a MUI implementation detail
  });
});
