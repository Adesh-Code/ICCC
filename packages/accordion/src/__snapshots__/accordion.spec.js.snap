// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion mounts an accordion 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAccordion-root  DhigAccordion--variant--individualPanel DhigAccordion--type--accordionList MuiAccordion-rounded MuiAccordion-gutters emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body-copy-medium dhig-m-0 emotion-1"
    >
      Accordion content
    </p>
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-2"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical emotion-3"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-4"
        >
          <div
            class="MuiAccordion-region"
            role="region"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
